extra/build_conda_packed.sh

```diff
@@ -1,78 +1,27 @@
 #!/bin/bash
 
-ENV_NAME="isolated-base"
-PYTHON_VERSION="3.11"
-PKGNAME="refl1d"
-SUBNAME="packed"
-OUTPUT="artifacts"
+# Definitions
+OUTPUT="conda_packed"
 SCRIPT_DIR=$(realpath $(dirname "${BASH_SOURCE[0]}"))
 SRC_DIR=$(dirname "$SCRIPT_DIR")
-cd $SRC_DIR
-
-mkdir -p $OUTPUT
 
 eval "$(conda shell.bash hook)"
-conda activate base || { echo 'failed: conda not installed'; exit 1; }
 
-conda install -y conda-pack
-if ! test -f "$ENV_NAME.tar.gz"; then
-  echo "creating isolated environment"
-  conda remove -n "$ENV_NAME" -y --all
-  conda create -n "$ENV_NAME" -q --force -y "python=$PYTHON_VERSION" "nodejs"
-  conda-pack -n "$ENV_NAME" -f -o "$ENV_NAME.tar.gz"
+if ! command -v conda-pack &> /dev/null; then
+  conda install -y conda-pack
 fi
 
-# unpack the new environment, that contains only python + pip
-tmpdir=$(mktemp -d)
-pkgdir="$tmpdir/$PKGNAME"
-envdir="$pkgdir/env"
-mkdir -p "$envdir"
-tar -xzf "$ENV_NAME.tar.gz" -C "$envdir"
-
-# activate the unpacked environment and install pip packages
-# add our batch script:
-case $OSTYPE in 
-  darwin*) cp -r ./extra/platform_scripts/refl1d_webview.app "$pkgdir" ;
-           cp -r ./extra/platform_scripts/refl1d_shell.app "$pkgdir" ;;
-  msys*) cp ./extra/platform_scripts/refl1d_webview.bat "$pkgdir" ;
-         cp ./extra/platform_scripts/refl1d_shell.bat "$pkgdir" ;;
-  linux*) cp -r ./extra/platform_scripts/make_linux_desktop_shortcut.sh "$pkgdir" ;
-          cp -r ./extra/platform_scripts/refl1d-webview "$pkgdir" ;;
-esac
-
-case "$OSTYPE" in
- "msys") bindir=$envdir ;
-         platform="Windows";;
- *) bindir=$envdir/bin ;
-    platform="$(uname -s)";;
-esac
-
-mkdir -p $pkgdir/share/icons
-cp $SCRIPT_DIR/*.svg $pkgdir/share/icons
-cp $SCRIPT_DIR/*.png $pkgdir/share/icons
-
-# install the packages
-$bindir/python -m pip install --no-input --no-compile numba
-# base path to source is in parent of SCRIPT_DIR
-$bindir/python -m pip install --no-input --no-compile .[webview]
-$bindir/python -m pip install orsopy
+ISOLATED_ENV="$(mktemp -d)/env"
+conda create -y -p "$ISOLATED_ENV" "python=${PYTHON_VERSION:-3.12}" "nodejs" "micromamba" "pip"
 
-cd $tmpdir
-# build the client
-$bindir/python -m refl1d.webview.build_client --cleanup
+cd $SCRIPT_DIR
+conda activate "$ISOLATED_ENV"
 
-version=$($bindir/python -c "import refl1d; print(refl1d.__version__)")
-mv "$tmpdir/$PKGNAME" "$tmpdir/$PKGNAME-$version"
+python -m pip install --no-input --no-compile "..[webview]"
```

Comment:
Presumably this runs versioningit so that the version number in the bundle is correct?

Reply:
Yes, I think versioningit is triggered during a pip install.  But I should check.